sdk/python/feast/importer.py

```diff
@@ -1,28 +1,47 @@
 import importlib
+from typing import Optional
 
-from feast import errors
+from feast.errors import (
+    FeastClassImportError,
+    FeastClassInvalidName,
+    FeastModuleImportError,
+)
 
 
-def get_class_from_type(module_name: str, class_name: str, class_type: str):
-    if not class_name.endswith(class_type):
-        raise errors.FeastClassInvalidName(class_name, class_type)
+def get_class_from_module(
+    module_name: str, class_name: str, class_type: Optional[str] = None
+):
+    """
+    Dynamically loads and returns a class from a module.
 
-    # Try importing the module that contains the custom provider
+    Args:
+        module_name: The name of the module.
+        class_name: The name of the class.
+        class_type: Optional suffix of the class.
+
+    Raises:
+        FeastClassInvalidName: If the class name does not end with the specified suffix.
+        FeastModuleImportError: If the module cannot be imported.
+        FeastClassImportError: If the class cannot be imported.
+    """
+    if class_type and not class_name.endswith(class_type):
```

Comment:
alternatively, we can make `class_type` something like base class expectations, eg we importing class FileStorage and expect that it's subtype of Storage.
Check base classes of imported type instead of naming could look like this
```
if not any(base_class.__name__ in class_type for base_class _class.__bases__):
    raise 
```